studio-android/LightNovelLibrary/app/src/main/java/org/mewx/wenku8/fragment/FavFragment.java

```diff
@@ -171,8 +171,34 @@ public void onPositive(MaterialDialog dialog) {
 
     @Override
     public void onItemLongClick(View view, int position) {
-        // Toast.makeText(getActivity(),"item long click detected", Toast.LENGTH_SHORT).show();
+       new MaterialDialog.Builder(getActivity())
```

Comment:
This is not a good way for interaction, because long click should provided a set of functions, instead of just deleting cache.

I reckon a better way is to add a pop-up menu for `selecting deleting from bookshelf` or `deleting local cache only`. In this way, the top-right dustbin button for each book in the bookshelf could be removed accordingly.